pkg/operator/staticpod/certsyncpod/certsync_controller.go

```diff
@@ -298,6 +300,21 @@ func (c *CertSyncController) Run(workers int, stopCh <-chan struct{}) {
 	}
 	klog.V(2).Infof("CertSyncer caches synced")
 
+	if len(c.cachesSyncedFile) != 0 {
+		// Signal caches sync using a file (e.g. for readiness probe)
+		f, err := os.OpenFile(c.cachesSyncedFile, os.O_APPEND|os.O_CREATE|os.O_RDONLY, 0600)
+		if err != nil {
+			klog.Fatal(err)
+		}
+
+		klog.V(1).Infof("CertSyncer caches synced file %q created.", c.cachesSyncedFile)
+
+		err = f.Close()
```

Comment:
defer f.Close()

Reply:
actually, nvmd.

Reply:
you don't check the error that way (unless you wrap it in a lambda) and there isn't a place for branching in between to escape the flow